src/project_context.rs

```diff
@@ -91,15 +91,36 @@ impl Context {
         })
     }
 
+    pub fn workspace_dir(&self) -> Result<PathBuf> {
+        let mut path = self.project_path.clone();
+        if let Some(workspace_dir) = self.config.rust.workspace_dir.as_ref() {
+            match workspace_dir.to_str() {
+                Some(".") => {}
+                Some(CONTRACTS_DIR) => {
+                    path.push(workspace_dir);
+                }
+                dir => {
+                    return Err(anyhow!("Invalid `workspace_dir` config: {:?}, only allowed \".\" or \"contracts\".", dir));
```

Comment:
Why? We have no reason to prevent us from accepting more paths.

https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.canonicalize